lib/ts/unit-tests/test_MemArena.cc

```diff
@@ -53,45 +58,77 @@ TEST_CASE("MemArena freeze and thaw", "[libts][MemArena]")
   MemSpan span1{arena.alloc(1024)};
   REQUIRE(span1.size() == 1024);
   REQUIRE(arena.size() == 1024);
+  REQUIRE(arena.reserved_size() >= 1024);
 
   arena.freeze();
 
   REQUIRE(arena.size() == 0);
   REQUIRE(arena.allocated_size() == 1024);
-  REQUIRE(arena.extent() >= 1024);
+  REQUIRE(arena.reserved_size() >= 1024);
 
   arena.thaw();
   REQUIRE(arena.size() == 0);
-  REQUIRE(arena.extent() == 0);
+  REQUIRE(arena.allocated_size() == 0);
+  REQUIRE(arena.reserved_size() == 0);
 
-  arena.reserve(2000);
+  span1 = arena.alloc(1024);
+  arena.freeze();
+  auto extent{arena.reserved_size()};
   arena.alloc(512);
-  arena.alloc(1024);
-  REQUIRE(arena.extent() >= 1536);
-  REQUIRE(arena.extent() < 3000);
-  auto extent = arena.extent();
+  REQUIRE(arena.reserved_size() > extent); // new extent should be bigger.
+  arena.thaw();
+  REQUIRE(arena.size() == 512);
+  REQUIRE(arena.reserved_size() >= 1024);
+
+  arena.clear();
```

Comment:
It will be the size of all allocated memory. That's the size used normally  for selecting the next reserved chunk. I think I"ll add a hint to `clear` for that reason.

Reply:
OK let me see if I have this right:

* If we reserved _R_ bytes, then allocated _A_ < _R_ bytes (in sum) from that reservation, then call `clear()`, then call `alloc(0)`, how many bytes will be reserved?

  **Exactly _R_-_A_ bytes are reserved?**

* If we reserved _R_ bytes, then allocated _A_ > _R_ bytes (in sum; where _A_ - _R_ is less than a memory page, for simplicity) how many bytes will be reserved?

  **_R_ bytes + new a block of _S_ bytes are reserved, where _S_ >= _R_?**


Generalizing:

* When we do a `freeze` or a `clear`, the next block is as big as the **sum of what was allocated**.

* When we `alloc` more bytes (in sum) than the size of the reservation, the new block that is reserved is **at least as many bytes as was allocated** in the original block.

Reply:
* The reservation will be large enough to hold A bytes.
* R bytes in the existing reservation plus S reserved bytes such that S > A.
* Yes, the next reservation is by default large enough to hold allocated memory.
* No, the new block is large enough to hold all allocated memory, which can be in more than one block. E.g. the third reserved block is large enough to hold all the memory allocated in the first and second blocks To quote from the docs, "The default style for reservation sizing follows a Fibonacci pattern, yielding (asymptotically) a block size growth rate of phi == 1.1618... which is close to the square root of 2.".

Reply:
Thanks,

> The reservation will be large enough to hold A bytes.

* Then the code comment was wrong for `clear`. But I see this has been updated to refer to a hint, and it seems OK now.

> R bytes in the existing reservation plus S reserved bytes such that S > A.

* Agree

> Yes, the next reservation is by default large enough to hold allocated memory.

* OK

> No, the new block is large enough to hold all allocated memory, which can be in more than one block. 

* My statement is true I think for 2 blocks (assumed), and false for more blocks than that.

  If I am not mistaken, your statement is true for 2 blocks, true for 3 blocks, and false for more than that. The *n*th Fibonacci number is *F(n)* = *F(n-1)*+*F(n-2)*. (E.g.: Fibonacci starts like `1 1 2 3 5`. The 5th number `5` is not greater than the sum of previous numbers `1+1+2+3=7`, so its block would not be large enough to hold all allocated memory.)

  I see this is a problem because we `freeze` says that the new block will by default be large enough to hold all the memory that has been allocated so far. But this is false when there are more than 2 internal blocks that have been frozen.

Reply:
No, because the allocated memory is tracked explicitly and globally, not per block. Therefore when `frozen()` is called the hint is the total allocated memory.

Reply:
OK, reference to the Fibonacci behavior was removed, so this is consistent now.